pkg/controller/hostpathprovisioner/controller.go

```diff
@@ -277,7 +277,11 @@ func (r *ReconcileHostPathProvisioner) Reconcile(context context.Context, reques
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
-	reqLogger.Info("Reconciling CSI and legacy controller plugin")
+	if r.isLegacy(cr) {
+		reqLogger.Info("Detected legacy CR, Reconciling CSI and legacy controller plugin")
```

Comment:
title says `Only start legacy provisioner on legacy CR` but here if you detect legacy,
will reconcile both CSI & legacy controller?

Reply:
Yes, the CSI driver is started always, but it only starts the legacy provisioner if the legacy CR is detected. This is mainly to facilitate upgrades.